src/dvc_render/vega.py

```diff
@@ -44,16 +39,15 @@ def __init__(self, datapoints: List, name: str, **properties):
 
     def get_filled_template(
         self, skip_anchors: Optional[List[str]] = None, strict: bool = True
-    ) -> str:
+    ) -> Dict[str, Any]:
         """Returns a functional vega specification"""
+        self.template.reset()
         if not self.datapoints:
-            return ""
+            return {}
 
         if skip_anchors is None:
             skip_anchors = []
 
-        content = deepcopy(self.template.content)
```

Comment:
Without deep-copying, we will be modifying the same contents.

Reply:
> Without deep-copying, we may accidentally modify the same contents.

I added ` reset()` method to bypass the need of deepcopy

Reply:
That’s only protected because replace_value copies the whole collection, right?